app/addons/components/react-components.react.jsx

```diff
@@ -1046,13 +1055,28 @@ function (app, FauxtonAPI, React, ReactDOM, Stores, Actions,
     },
 
     getDocContent: function () {
-      if (!_.isEmpty(this.props.docContent)) {
-        return (
-          <div className="doc-data">
-            <pre className="prettyprint">{this.props.docContent}</pre>
-          </div>
-        );
+      if (_.isEmpty(this.props.docContent)) {
+        return null;
+      }
+
+      // if need be, truncate the document
+      var content = this.props.docContent;
+      var isTruncated = false;
+      if (this.props.truncate) {
+        var lines = this.props.docContent.split('\n');
+        if (lines.length > this.props.maxRows) {
+          isTruncated = true;
+          lines = lines.slice(0, this.props.maxRows);
+          content = lines.join('\n');
+        }
```

Comment:
Yeah, this feels like the wrong place, hey. Almost feels like a helper function. I'll move it.